java/src/main/java/ai/rapids/cudf/MemoryBuffer.java

```diff
@@ -240,15 +240,18 @@ public synchronized void close() {
     if (cleaner != null) {
       refCount--;
       cleaner.delRef();
-      if (eventHandler != null) {
-        eventHandler.onClosed(refCount);
-      }
-      if (refCount == 0) {
-        cleaner.clean(false);
-        closed = true;
-      } else if (refCount < 0) {
-        cleaner.logRefCountDebug("double free " + this);
-        throw new IllegalStateException("Close called too many times " + this);
+      try {
+        if (refCount == 0) {
+          cleaner.clean(false);
+          closed = true;
+        } else if (refCount < 0) {
+          cleaner.logRefCountDebug("double free " + this);
+          throw new IllegalStateException("Close called too many times " + this);
+        }
+      } finally {
+        if (eventHandler != null) {
+          eventHandler.onClosed(refCount);
+        }
       }
```

Comment:
This is now inconsistent with (Host)ColumnVector. If we cannot consistently move eventHandler.onClosed into a finally please add a comment, but hopefully we can and it could be generalized via some utls withEventHandler(eventHandler) or base class implementation.

Reply:
good catch, I will make them the same, but won't introduce a new interface right now. I'll file a follow on for that. 

I don't see uses of onClosed when refCount == 0 in spill framework, really only on `HostAlloc` code paths, which is `MemoryBuffer` specific.

Reply:
https://github.com/rapidsai/cudf/issues/15354

Reply:
@gerashegalov pushed https://github.com/rapidsai/cudf/pull/15351/commits/446ab9bc651cac224da7551336aeda26c2a2e4b2.

Reply:
@abellina my comment referred to this as well 
https://github.com/rapidsai/cudf/blob/f9ac4277f50163a7da2006460034aa3e45c8744e/java/src/main/java/ai/rapids/cudf/HostColumnVector.java#L150-L165

Reply:
oh gosh, we really ought to get that interface you requested. Thanks, fixing.

Reply:
https://github.com/rapidsai/cudf/pull/15351/commits/31e5be31527025bdf0ce0d6c7799e0047b2d1e13 @gerashegalov